File: OC/内容层动画/WGOCCoreAnimation/WGOCCoreAnimation/CAShapeLayer动画/WGChartAniViewController.m
//
//  WGChartAniViewController.m
//  WGOCCoreAnimation
//
//  Created by wanggang on 2019/5/27.
//  Copyright © 2019 bozhong. All rights reserved.
//

#import "WGChartAniViewController.h"
#import "WGLineChartView.h"
#import "WGBarChartView.h"

@interface WGChartAniViewController ()

@property (nonatomic, strong) WGLineChartView *lineView;
@property (nonatomic, strong) WGBarChartView *barView;

@end

@implementation WGChartAniViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"图表动画";
    self.view.backgroundColor = [UIColor blackColor];

    self.lineView = [[WGLineChartView alloc] initWithFrame:CGRectMake(0, 0, SCREEN_WIDTH, SCREEN_HEIGHT/2)];
    [self.view addSubview:_lineView];
    
    self.barView = [[WGBarChartView alloc] initWithFrame:CGRectMake(0, _lineView.bottom, SCREEN_WIDTH, SCREEN_HEIGHT/2-50)];
    [self.view addSubview:_barView];
    
    UIButton *btn = [[UIButton alloc] initWithFrame:CGRectMake(0, _barView.bottom, SCREEN_WIDTH, 50)];
    [btn setTitle:@"点我" forState:UIControlStateNormal];
    [btn setBackgroundColor:UIColor.yellowColor];
    [btn setTitleColor:UIColor.blackColor forState:UIControlStateNormal];
    [btn addTarget:self action:@selector(btnClicked) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:btn];
    
}

- (void)btnClicked{
    
    _lineView.pointArr = @[
                           [NSValue valueWithCGPoint:CGPointMake(30+35, 260)],
                           [NSValue valueWithCGPoint:CGPointMake(100+35, 100)],
                           [NSValue valueWithCGPoint:CGPointMake(170+35, 240)],
                           [NSValue valueWithCGPoint:CGPointMake(240+35, 170)],
                           [NSValue valueWithCGPoint:CGPointMake(310+35, 220)]
                           ];
    _lineView.lineColor = UIColor.greenColor;
    [_lineView startAni];
    
    _barView.pointArr = @[
                          @[
                              [NSValue valueWithCGPoint:CGPointMake(60, 215)],
                              [NSValue valueWithCGPoint:CGPointMake(60, 100)],
                              ],
                          @[
                              [NSValue valueWithCGPoint:CGPointMake(130, 215)],
                              [NSValue valueWithCGPoint:CGPointMake(130, 120)],
                              ],
                          @[
                              [NSValue valueWithCGPoint:CGPointMake(200, 215)],
                              [NSValue valueWithCGPoint:CGPointMake(200, 140)],
                              ],
                          @[
                              [NSValue valueWithCGPoint:CGPointMake(270, 215)],
                              [NSValue valueWithCGPoint:CGPointMake(270, 160)],
                              ],
                          @[
                              [NSValue valueWithCGPoint:CGPointMake(340, 215)],
                              [NSValue valueWithCGPoint:CGPointMake(340, 180)],
                              ],
                          ];
    _barView.lineColor = UIColor.purpleColor;
    [_barView startAni];
}



@end

File: Swift/内容层动画/WGCoreAnimation/WGCoreAnimation/CAShapeLayer动画/WGChartAniViewController.swift
//
//  WGChartAniViewController.swift
//  WGCoreAnimation
//
//  Created by wanggang on 2019/5/7.
//  Copyright © 2019 wanggang. All rights reserved.
//

import UIKit

class WGChartAniViewController: UIViewController {

    var lineChart: WGLineChartView!
    var barChart: WGBarChartView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = UIColor.white
        title = "图表动画"
        
        lineChart = WGLineChartView(frame: self.view.bounds)
        barChart = WGBarChartView(frame: CGRect(x: 0, y: WgHeight/2, width: WgWith, height: WgHeight))
        view.addSubview(lineChart)
        view.addSubview(barChart)
        buildView()
    }
    
    func buildView() {
        
        let width: CGFloat = 70
        let height: CGFloat = 20
        
        for i in 1...5 {
            let title = "value"+"\(i)"
            let label = UILabel(frame: CGRect(x: 30+(CGFloat(i)-1)*70, y: 300, width: width, height: height))
            label.text = title
            view.addSubview(label)
        }
        
        for i in 0...5{
            let title = "\(10-i*2)"
            let label = UILabel(frame: CGRect(x: 5,y: 120+(CGFloat(i)-1)*35, width: 20, height: 20))
            label.text = title
            self.view.addSubview(label)
        }
        
        let lineBtn = UIButton(frame: CGRect(x: (WgWith-100)/2, y: 320, width: 100, height: 30))
        lineBtn.setTitle("开始", for: .normal)
        lineBtn.setTitleColor(UIColor.purple, for: .normal)
        lineBtn.backgroundColor = UIColor.yellow
        lineBtn.addTarget(self, action: #selector(lineBtnClicked), for: .touchUpInside)
        view.addSubview(lineBtn)
        
        //      BarChart
        for i in 1...5{
            let xAxesTitle:String = "SEP"+"\(i)"
            let xAxesLabel:UILabel = UILabel(frame: CGRect(x: 40+(CGFloat(i)-1)*70,y: 600, width: 50, height: 20))
            xAxesLabel.text = xAxesTitle
            self.view.addSubview(xAxesLabel)
        }
    }
    
    @objc func lineBtnClicked() {
        
        lineChart.chartAniStart()
        barChart.barAniStart()
    }
}
